web-ng/src/app/components/main-page/main-page.component.ts

```diff
@@ -86,9 +92,24 @@ export class MainPageComponent implements OnInit {
     }
   }
 
-  private showEditLayerDialog(layerId: string) {
-    this.dialog.open(LayerDialogComponent, {
-      data: { layerId },
+  private showEditLayerDialog(layerId?: string) {
+    this.activeProject$.pipe(take(1)).subscribe(project => {
+      let layerDialogData = {};
+      if (layerId) {
```

Comment:
No conditional needed here; you can just set `createNew: layerId == ":new"` in the map.  The local vars could then be removed and the logic inlined below in the call to `dialog.open()`.